cdap-ui/app/cdap/components/AbstractWidget/KeyValueDropdownWidget/KeyValueDropdownRow.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import Input from '@material-ui/core/Input';
+import withStyles from '@material-ui/core/styles/withStyles';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import AbstractRow, {
+  IAbstractRowProps,
+} from 'components/AbstractWidget/AbstractMultiRowWidget/AbstractRow';
+
+const styles = (theme) => {
+  return {
+    root: {
+      height: '44px',
+    },
+    inputContainer: {
+      width: 'calc(100% - 100px)',
```

Comment:
We should try to avoid using `calc` anymore.
1. The `100px` is not in a variable
2. Changing one place affects another if we don't re-adjust the calc

I would recommend us trying to use `grid`. Nested grids (one for each row and one for the textbox + select dropdown). 
With grids we can specify in terms of `fr`s and the onus goes to the browser to calculate and readjust the width

Reply:
An example of [the idea](https://codepen.io/ajainarayanan/pen/OYGpxp)